tokio-macros/src/entry.rs

```diff
@@ -70,11 +79,17 @@ impl Configuration {
         if self.flavor.is_some() {
             return Err(syn::Error::new(span, "`flavor` set multiple times."));
         }
-
-        let runtime_str = parse_string(runtime, span, "flavor")?;
-        let runtime =
-            RuntimeFlavor::from_str(&runtime_str).map_err(|err| syn::Error::new(span, err))?;
-        self.flavor = Some(runtime);
+        # [cfg(not(target_os = "wasi"))]
+        {
+            let runtime_str = parse_string(runtime, span, "flavor")?;
+            let runtime =
+                RuntimeFlavor::from_str(&runtime_str).map_err(|err| syn::Error::new(span, err))?;
+            self.flavor = Some(runtime);
+        }
+        #[cfg(target_os = "wasi")]
+        {
+            self.flavor = Some(RuntimeFlavor::CurrentThread);
+        }
```

Comment:
As far as I can tell, you're just ignoring the macro options here. We shouldn't do that. If somebody specifies `#[tokio::main(flavor = "multi_thread")]` (as opposed to not specifying any flavor and using the default), then we should emit an error message saying that threading is not available on WASM. The macro already has support for detecting the difference between "use the default" and "explicitly set the option to its default value".